esp/esdllib/esdl_def.cpp

```diff
@@ -1590,10 +1602,22 @@ IEsdlDefObjectIterator* EsdlDefinition::getDependencies( const char* service, St
     this->gatherMethodDependencies( *dependencies, methodArray, requestedVer, opts, flags );
 
     bool allTypes = !(flags & DEPFLAG_INCLUDE_TYPES); //not asking for any explicit types indicates all types
-    if(serviceDef && (allTypes || (flags & DEPFLAG_INCLUDE_SERVICE)))
+    if (allTypes || (flags & DEPFLAG_INCLUDE_SERVICE))
     {
-        EsdlDefObjectWrapper* wrapper = new EsdlDefObjectWrapper( *serviceDef );
-        dependencies->append( *wrapper );
+        if (serviceDef)
+        {
+            EsdlDefObjectWrapper* wrapper = new EsdlDefObjectWrapper( *serviceDef );
+            dependencies->append( *wrapper );
```

Comment:
not directly related to this PR, but I notice 'dependencies' would leak (line 1556) if an exception was thrown. More uses of Owned / getClear() etc. would be good.

Reply:
EsdlDefObjectWrapperArray is not inherited from Interface so Owned won't work. I'll create a simple conditional releasing class to deal with this scenario.

Reply:
Actually jlib's OwnedPtr class is perfect for this scenario.